lake/segment/writer.go

```diff
@@ -1,76 +1,67 @@
 package segment
 
 import (
+	"bytes"
 	"context"
 	"io"
 
+	"github.com/brimdata/zed/field"
 	"github.com/brimdata/zed/lake/seekindex"
 	"github.com/brimdata/zed/order"
 	"github.com/brimdata/zed/pkg/bufwriter"
-	"github.com/brimdata/zed/pkg/nano"
 	"github.com/brimdata/zed/pkg/storage"
-	"github.com/brimdata/zed/zio"
 	"github.com/brimdata/zed/zio/zngio"
 	"github.com/brimdata/zed/zng"
 )
 
 // Writer is a zbuf.Writer that writes a stream of sorted records into a
 // data segment.
 type Writer struct {
-	ref           *Reference
-	byteCounter   *writeCounter
-	count         uint64
-	size          int64
-	rowObject     *zngio.Writer
-	firstTs       nano.Ts
-	lastTs        nano.Ts
-	needSeekWrite bool
-	order         order.Which
-	seekIndex     *seekindex.Builder
-	wroteFirst    bool
+	ref             *Reference
+	byteCounter     *writeCounter
+	count           uint64
+	rowObject       *zngio.Writer
+	firstKey        zng.Value
+	lastKey         zng.Value
+	needSeekWrite   bool
+	order           order.Which
+	seekIndex       *seekindex.Writer
+	seekIndexCloser io.Closer
+	first           bool
+	poolKey         field.Path
 }
 
-type WriterOpts struct {
-	Order order.Which
-	Zng   zngio.WriterOpts
-}
-
-func (r *Reference) NewWriter(ctx context.Context, engine storage.Engine, path *storage.URI, opts WriterOpts) (*Writer, error) {
+func (r *Reference) NewWriter(ctx context.Context, engine storage.Engine, path *storage.URI, o order.Which, seekIndexFactor int, poolKey field.Path) (*Writer, error) {
 	out, err := engine.Put(ctx, r.RowObjectPath(path))
 	if err != nil {
 		return nil, err
 	}
 	counter := &writeCounter{bufwriter.New(out), 0}
-	writer := zngio.NewWriter(counter, opts.Zng)
-	seekObjectPath := r.SeekObjectPath(path)
-	seekIndex, err := seekindex.NewBuilder(ctx, engine, seekObjectPath.String(), opts.Order)
+	opts := zngio.WriterOpts{
+		StreamRecordsMax: seekIndexFactor,
+		LZ4BlockSize:     zngio.DefaultLZ4BlockSize,
+	}
+	writer := zngio.NewWriter(counter, opts)
```

Comment:
```suggestion
	writer := zngio.NewWriter(counter, zngio.WriterOpts{
		StreamRecordsMax: seekIndexFactor,
		LZ4BlockSize:     zngio.DefaultLZ4BlockSize,
	})
```

Reply:
The problem here is the design pattern for calling NewWriter with opts.